bin/brew

```diff
@@ -45,5 +85,77 @@ fi
 
 export HOMEBREW_BREW_FILE
 export HOMEBREW_RUBY_PATH
+export HOMEBREW_PREFIX
+export HOMEBREW_REPOSITORY
+export HOMEBREW_LIBRARY
+export HOMEBREW_CELLAR
+
+if [ -n "$HOMEBREW_OSX" ]
+then
+  if [ "$(xcode-\select --print-path)" = "/" ]
+  then
+    cat >&2 <<EOS
+Your xcode-select path is currently set to '/'.
+This causes the 'xcrun' tool to hang, and can render Homebrew unusable.
+If you are using Xcode, you should:
+  sudo xcode-select -switch /Applications/Xcode.app
+Otherwise, you should:
+  sudo rm -rf /usr/share/xcode-select
+EOS
+    exit 1
+  fi
+
+  XCRUN_OUTPUT="$(/usr/bin/xcrun clang 2>&1)"
+  XCRUN_STATUS="$?"
+
+  if [ "$XCRUN_STATUS" != "0" ] && [[ "$XCRUN_OUTPUT" =~ license ]]
```

Comment:
Two things:

- I think it is a bit less confusing to check the exit status with a numerical comparison, i.e., `"$XCRUN_STATUS" -ne 0` instead of `"$XCRUN_STATUS" != "0"`.

- The second check doesn't really require a regular expression. Simpler and probably faster is `[[ "$XCRUN_OUTPUT" = *license* ]]`.